src/types/path.ts

```diff
@@ -0,0 +1,11 @@
+import { Param } from '@/types/params'
+
+export type PathParams = Record<string, Param[]>
+
+export type Path<
+  T extends string = any,
+  P extends PathParams = any
+> = {
+  path: T,
+  params: Required<P>,
```

Comment:
This type of the return will need to reflect that it actually returns the MergedParams because we'll need that type to fix the route method types.